modules/tinymce/src/core/main/ts/html/Sanitization.ts

```diff
@@ -240,17 +240,20 @@ const sanitizeMathmlElement = (node: Element, settings: DomParserSettings) => {
   };
 
   purify.addHook('uponSanitizeElement', (node, evt) => {
-    const lcTagName = evt.tagName ?? node.nodeName.toLowerCase();
+    // We know the node is an element as we have
+    // passed an element to the purify.sanitize function below
+    const elm = node as Element;
```

Comment:
This is not really correct. The `uponSanitizeElement` is oddly named and the node is actually text nodes, comments and elements. I should have pointed that out on the original the type is incorrect in the original but since we are checking the tag name it kind of safe to pass it in as a element to hasValidEncoding so maybe we can cast it there instead.

Reply:
Thanks for the clarification Spocke. This was an incorrect assumption on my part. I'll fix this up